### PR TITLE
ARROW-7519: [Python] Build wheels, conda packages with dataset support

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by kszucs** [comment]
@ursabot crossbow submit -g conda -g wheel

File: dev/tasks/conda-recipes/arrow-cpp/bld.bat [hunk 1]
```diff
@@ -20,6 +20,7 @@ cmake -G "Ninja" ^
       -DCMAKE_BUILD_TYPE=release ^
       -DARROW_PYTHON:BOOL=ON ^
       -DARROW_MIMALLOC:BOOL=ON ^
+      -DARROW_DATASET:BOOL=ON ^
```
Comment (pitrou) [inline dev/tasks/conda-recipes/arrow-cpp/bld.bat]: Should also add ARROW_CSV, ARROW_COMPUTE, ARROW_FILESYSTEM and ARROW_JSON explicitly, no?

**Review by pitrou — CHANGES_REQUESTED** [review]
LGTM, but should enable other components as well

File: dev/tasks/conda-recipes/arrow-cpp/bld.bat (hunk 1 above)
Comment (wesm) [reply dev/tasks/conda-recipes/arrow-cpp/bld.bat]: These non-optional (for Python) components are enabled automatically when ARROW_PYTHON=ON

https://github.com/apache/arrow/blob/master/cpp/CMakeLists.txt#L286

**Comment by kszucs** [comment]
@ursabot crossbow submit -g conda -g wheel

**Review by pitrou — APPROVED** [review]

**Comment by pitrou** [comment]
@kszucs In https://travis-ci.org/ursa-labs/crossbow/builds/637010906#L7191, it seems that `test_dataset.py` relies on the order of dataset scanning results. This is wrong: datasets can be scanned in any order, AFAIK.

**Comment by kszucs** [comment]
@ursabot crossbow submit -g conda -g wheel

**Review by kszucs — COMMENTED** [review]
+1

**Pull request closed without merging**